### DISCUSSION TITLE
HandleSubmit is no longer triggered when deleting an array item with a validation error

### DISCUSSION THREAD

**Discussion by nino-cast** [react-hook-form/react-hook-form#7170] [Q&A] [answered]
### not working

1. Add input item
2. input invalid value
3. remove input item
4. try submit, but not working

### working

1. Add input item
2. input valid value
3. remove input item
4. try submit, but not working

---

In both cases isValid is true but handleSubmit is not triggered in the first case.

Works with clearErros in combination with the remove method

```tsx
<button type="button" onClick={() => remove(i); clearErrors(`links.${i}`)}>
  remove
</button>
```

Is it a bug that handleSubmit is not triggered when isValid is true?

Thank you for your help.

## Code

https://codesandbox.io/s/elated-ishizaka-womz3?file=/src/App.js

**Comment by bluebill1049** [comment]
this is a valid bug.

**Comment by bluebill1049** [comment, marked as answer]
thanks for reporting this issue, release this fix now.

**Reply by nino-cast** [reply]
Thank you for your quick response! 🎉🎉🎉🎉
